### PR TITLE
TFM: configurable memory layout

### PR DESCRIPTION
**Pull request by oyvindronningstad**
Only the top-level CMakeLists.txt needs to be reviewed.

### CODE REVIEW

**Review by ioannisg — APPROVED** [review]

**Review by hakonfam — APPROVED** [review]

File: trusted-firmware-m/CMakeLists.txt [hunk 1]
```diff
@@ -11,6 +11,13 @@ set(TFM_VERSION 1.2.0)
 
 ############################ CONFIGURATION #####################################
 
+if (IS_ABSOLUTE "${TFM_PLATFORM}")
```
Comment (tejlmand) [inline trusted-firmware-m/CMakeLists.txt:14]: Nothing wrong here, but the commit message states:
> Synchronize with commit a9d5eacdb045a5d793aca0e591ec3abe48fa7205

but I don't find this commit anywhere. 
Could you provide more info ?

Also, are all changes in this commit 319fcf6331fc2e260b0bc28c9f81653ead7a21eb to be reverted once commit a9d5eacdb045a5d793aca0e591ec3abe48fa7205 is merged somewhere ?

File: tf-m-tests/app/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #-------------------------------------------------------------------------------
-# Copyright (c) 2020, Arm Limited. All rights reserved.
+# Copyright (c) 2020-2021, Arm Limited. All rights reserved.
```
Comment (tejlmand) [inline tf-m-tests/app/CMakeLists.txt:2]: Same with this commit: 
> Synchronize with commit 402a298e2eb32dc1c72d991eacf300bc96e78416

File: trusted-firmware-m/CMakeLists.txt (hunk 1 above)
Comment (oyvindronningstad) [reply trusted-firmware-m/CMakeLists.txt:14]: It's upstream TFM. https://review.trustedfirmware.org/plugins/gitiles/TF-M/trusted-firmware-m/+/a9d5eacdb045a5d793aca0e591ec3abe48fa7205

File: tf-m-tests/app/CMakeLists.txt (hunk 2 above)
Comment (oyvindronningstad) [reply tf-m-tests/app/CMakeLists.txt:2]: It's upstream tf-m-tests. https://review.trustedfirmware.org/plugins/gitiles/TF-M/tf-m-tests/+/402a298e2eb32dc1c72d991eacf300bc96e78416

File: trusted-firmware-m/CMakeLists.txt (hunk 1 above)
Comment (tejlmand) [reply trusted-firmware-m/CMakeLists.txt:14]: thanks.

**Review by tejlmand — APPROVED** [review]

**Pull request merged**